float32 red
float32 yellow
float32 green
float32 area